zeebe/gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/util/XmlUtil.java

```diff
@@ -48,21 +48,25 @@ private SAXParserFactory getSAXParserFactory() {
   }
 
   public void extractFlowNodeNames(
-      final Long processDefinitionKey, final BiConsumer<Long, ProcessFlowNode> flowNodeConsumer) {
+      final Long processDefinitionKey,
+      final BiConsumer<Long, ProcessFlowNode> processDefinitionKeyFlowNodeConsumer) {
     final var processDefinition = processDefinitionServices.getByKey(processDefinitionKey);
-    extractFlowNodeNames(processDefinition, flowNodeConsumer);
+    extractFlowNodeNames(processDefinition, processDefinitionKeyFlowNodeConsumer);
   }
 
   public void extractFlowNodeNames(
       final Set<Long> processDefinitionKeys,
-      final BiConsumer<Long, ProcessFlowNode> flowNodeConsumer) {
+      final BiConsumer<Long, ProcessFlowNode> processDefinitionKeyFlowNodeConsumer) {
     final var keysList = new ArrayList<>(processDefinitionKeys);
     final var result =
         processDefinitionServices.search(
-            ProcessDefinitionQuery.of(q -> q.filter(f -> f.processDefinitionKeys(keysList))));
+            ProcessDefinitionQuery.of(
+                q ->
+                    q.filter(f -> f.processDefinitionKeys(keysList))
+                        .page(p -> p.size(processDefinitionKeys.size()))));
```

Comment:
💭 It probably is never happening, but to name it here: we might still be limited to a subset of all results we want to get. I think the `size` is just a "hint" on how many results we'd like to receive, it can be smaller if the database has an internal limit that is smaller (e.g. we request 20k process definition keys but ES will only deliver 10k max).

Is this realistic? I don't know, probably not in this case here. Just wanted to leave this here still so we don't blindly run into this 😅

Reply:
This is why @houssain-barouni implemented `findAll` for [authorization search](https://github.com/camunda/camunda/blob/50d3a76829a966858a77cb2100ab87409d0d97d3/search/search-client-query-transformer/src/main/java/io/camunda/search/clients/SearchClients.java#L100).

Reply:
I think I will add the LOG for now. When we do the follow-up ticket with moving it into search clients, we can tackle this with a similar `findAll` approach.
What do you think @tmetzke?

Reply:
Yep sounds pragmatic to me 👍